specification/resources/resource-manager/Microsoft.Resources/stable/2021-04-01/resources.json

```diff
@@ -4605,6 +4608,50 @@
       ],
       "description": "Deployment properties."
     },
+    "DeploymentParameter": {
+      "properties": {
+        "value": {
+          "type": "object",
+          "description": "Input value to the parameter ."
+        },
+        "reference": {
+          "$ref": "#/definitions/KeyVaultReferenceParameter",
+          "description": "Azure Key Vault parameter reference."
+        },
+      },
+      "required": [
+      ],
+      "description": "Deployment parameter for the template."
+    },
+    "KeyVaultReferenceParameter": {
+      "properties": {
+        "keyVault": {
+          "$ref": "#/definitions/KeyVaultReference",
+          "description": "Input value to the parameter."
+        },
+        "secretName": {
```

Comment:
There's also an optional `secretVersion` property - see here for the server-side model we use: https://msazure.visualstudio.com/One/_git/AzureUX-Deployments?path=%2Fsrc%2FCore%2FEntities%2FKeyVaultParameterReference.cs&_a=contents&version=GBmaster

Reply:
Added the property and renamed the JSON model to match C#.